pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -587,21 +593,21 @@ func (p *PriorityQueue) Close() {
 	p.cond.Broadcast()
 }
 
-// DeleteNominatedPodIfExists deletes pod nominatedPods.
-func (p *PriorityQueue) DeleteNominatedPodIfExists(pod *v1.Pod) {
-	p.lock.Lock()
-	p.nominatedPods.delete(pod)
-	p.lock.Unlock()
+// DeleteNominatedPodIfExists deletes <pod> from nominatedPods.
+func (npm *NominatedPodMap) DeleteNominatedPodIfExists(pod *v1.Pod) {
+	npm.lock.Lock()
+	npm.delete(pod)
+	npm.lock.Unlock()
```

Comment:
`defer npm.lock.Unlock()`？

Reply:
I usually hold using defer unless it's necessary (more than 1 return path).